docs/proposals/multi-node/README.md

```diff
@@ -0,0 +1,623 @@
+This will mean that `etcd-backup-restore` becomes Kubernetes-aware. But there are other reasons for making `etcd-backup-restore` Kubernetes-aware anyway (e.g. to [cut off client requests](#cutting-off-client-requests-on-backup-failure)).
+This enhancement should keep `etcd-backup-restore` backward compatible.
+But it should be possible to use `etcd-backup-restore` Kubernetes-unaware as before this proposal. This is possible either by auto-detecting the existence of kubeconfig or by an explicit command-line flag (such as `--enable-etcd-status-updates` which can be defaulted to `false` for backward compatibility).
+
+### Note
+
+With approach [above](#status), members can be marked with `status: Ready` only by their `etcd-backup-restore` sidecar container.
+However, they can be marked with `status: NotReady` either by their `etcd-backup-restore` sidecar container (with `reason: HeartbeatFailed`) or by `etcd-druid` (as explained [below](#decision-table-for-etcd-druid-based-on-the-status)).
+
+### Alternative
+
+The alternative is for `etcd-druid` to maintain the status in the `Etcd` status sub-resource.
+But `etcd-druid` is centrally deployed in the host Kubernetes cluster and cannot scale well horizontally.
+So, it can potentially be a bottleneck if it is involved in regular health check mechanism for all the etcd clusters it manages.
+Also, the recommended approach above is more robust because it can work even if `etcd-druid` is down when the backup upload of a particular etcd cluster fails.
+
+## Decision table for etcd-druid based on the status
+
+The following decision table describes the various criteria `etcd-druid` takes in to consideration to detect different etcd cluster management scenarios and the corresponding reconciliation actions it must take.
+The general principle is to detect the scenario and take the minimum action to move the cluster along the path good health.
+The path from any one scenario to a state of good health will typically involve going through multiple reconciliation actions which probably take the cluster through many other cluter management scenarios.
+Especially, it is proposed that individual members auto-heal if where possible even in the case of the failure of a majority of members of the etcd cluster and that `etcd-druid` takes action only if the auto-healing doesn't happen for a configured period.
+
+############################ ETCD Cluster Management Scenario ############################ | Current Cluster Size | Desired Cluster Size | `StatefulSet` spec `replicas` | `StatefulSet` status `readyReplicas` | `Etcd` status members | `Etcd` status `Ready` members | `Etcd` status `NotReady` members with `lastTransitionTime > notReadyGracePeriod` | `Etcd` status `Ready` or `NotReady` members with `lastHeartbeatTime > heartbeatGracePeriod` | `Etcd` status `Unknown` members with `lastTransitionTime > unknownGracePeriod` | ######################################## Action ########################################
```

Comment:
Table not markdown-compliant (rendered only in GitHub), which will be also the reason why I had the other rendering issues locally (missing newlines).

Also, what's the purpose of the `###`s? Maybe it lacks `<br/>` to always render nicely, i.e. with the `###` in first and last row?

Reply:
I used ###s to achieve width for the columns as otherwise the columns were to narrow and wrapping multiple lines.